src/main/java/com/example/WorldOfAnimals/services/AnimalService.java

```diff
@@ -0,0 +1,54 @@
+package com.example.WorldOfAnimals.services;
+
+import com.example.WorldOfAnimals.dto.AnimalDTO;
+import com.example.WorldOfAnimals.dto.AnimalRequestDTO;
+import com.example.WorldOfAnimals.dto.AnimalRequestPutDTO;
+import com.example.WorldOfAnimals.exceptions.AnimalNotFoundException;
+import com.example.WorldOfAnimals.mapper.AnimalMapper;
+import com.example.WorldOfAnimals.models.AnimalEntity;
+import com.example.WorldOfAnimals.repositories.AnimalRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class AnimalService {
+
+    private final AnimalRepository repository;
+    private final AnimalMapper mapper;
+
+    @Transactional
+    public void saveAnimal(AnimalRequestDTO animalRequestDTO) {
+        repository.save(mapper.convertToEntityPost(animalRequestDTO));
+    }
+
+    @Transactional
```

Comment:
Why in get method you use Transaction, When method get, we can add   @Transactional(readOnly = true) on without  @Transactional

Reply:
done